test/common/runtime/filesystem_setup.sh

```diff
@@ -2,8 +2,11 @@
 
 set -e
 
+TEST_DATA=test/common/runtime/test_data
+
 cd ${TEST_RUNDIR}
-cp -rfL --parents test/common/runtime/test_data ${TEST_TMPDIR}
-chmod -R u+rwX ${TEST_TMPDIR}/test/common/runtime/test_data
-ln -sf ${TEST_TMPDIR}/test/common/runtime/test_data/root ${TEST_TMPDIR}/test/common/runtime/test_data/current
-ln -sf ${TEST_TMPDIR}/test/common/runtime/test_data/root/envoy/subdir ${TEST_TMPDIR}/test/common/runtime/test_data/root/envoy/badlink
+mkdir -p ${TEST_TMPDIR}/${TEST_DATA}
+cp -RfL ${TEST_DATA}/* ${TEST_TMPDIR}/${TEST_DATA}
```

Comment:
While you're here, can you quote all variable dereferences for path WSP safety, e.g. `+mkdir -p "${TEST_TMPDIR}/${TEST_DATA}"?